modules/grpc/bigquery/CMakeLists.txt

```diff
@@ -36,4 +36,4 @@ add_module(
 
 set_target_properties(bigquery PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib;${CMAKE_INSTALL_PREFIX}/lib/syslog-ng")
 
-target_compile_options (bigquery-cpp PRIVATE -Wno-switch-default -Wno-double-promotion -Wno-deprecated -Wno-inconsistent-missing-override -DPROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII=0)
+target_compile_options (bigquery-cpp PRIVATE -Wno-switch-default -Wno-switch-default -Wno-double-promotion -Wno-deprecated -Wno-inconsistent-missing-override -DPROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII=0)
```

Comment:
neither these, probably some of them also can be eliminated using the above-mentioned compatibility headers, but not all of them unfortunately

Reply:
is the switch coming from our code, or the library?

Reply:
as this will ignore the wrongness in our code instead of the dependencies

Reply:
> is the switch without the default branch coming from our code, or the library?

external code

Reply:
> as this will ignore the wrongness in our code instead of the dependencies

just like any of the other turned off warnings that is why I do not like these
but, not always can we do anything else
e.g. for -DPROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII=0
or in case of many other third-party header issues, we do not have control above

Reply:
we can just do the same as above, without ignoring warrnings in our code with
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignore
<includes>
#pragma GCC diagnostic pop
```

Reply:
exactly what i'd like to avoid, as I mentioned in the other comments, the code just started getting full of 4 liners like that, it is ugly, and makes the code unreadable more and more with each of those entries

my best bet is some kind of new compatibility headers, but any better hints are very welcome

Reply:
In my opinion it would be better than to use a compat header, than disabling warnings in our code not just in our dependencies. As some for example the mising default case for a switch warning is a typical mistake that can lead to some pretty unpleasant surprises over time.

Reply:
cmake changes moved to the upcoming grpc modification PRs